lib/tasks/upgrade.rake

```diff
@@ -0,0 +1,56 @@
+# TRANSLATORS: do not translate
+desc <<-END_DESC
+  This task runs predefined upgrade steps.
+
+  Examples:
+    rake upgrade:run
+    rake upgrade:mark_as_ran TASK_NAME=some_rake_task,another_rake_task
+
+END_DESC
+
+namespace :upgrade do
+  task :run => :environment do
+    ENV['FOREMAN_UPGRADE'] = '1'
+    tasks = Foreman::Plugin.registered_plugins.values.map{|plugin| plugin.upgrade_tasks}.flatten
```

Comment:
I generally find long chains hard to read.

Reply:
Using `.map(&:upgrade_tasks)` would shorten it already

Reply:
`.flat_map(&:upgrade_tasks)` will shorten it even more if it in only nested 1 level.